### ISSUE TITLE
Reimport breaks references to DCC Animations

### ISSUE THREAD

**Issue by lachsinc** [OurMachinery/themachinery-public#1346]
- **Version:** Git Master Branch | Latest Binary Release
- **OS:** Windows

Continuing on from the discussion in PR https://github.com/OurMachinery/themachinery/pull/985

Repro:
1. Create .clip that points to a DCC Animation
2. Reimport DCC Asset
3. Save, close, reopen
4. .clip reference broken

Affects the following:
- .clips 
- .animations
- Simple Animation Player Component
as they contain references to dcc animations.

Some related crashes have been addressed in the above PR but keeping this separate as the issue is more fundamental;

During `tm_dcc_asset_api->update()` the `retarget_write()` results in the original dcc_assets subobjects (all subobjects, not just animations) becoming "detached" objects. 
These objects continue to live in the truth until the machinery is restarted. 
Prior to restart, .clips, .animations etc. continue to refer to and (possibly?) access the stale data.
After restarting the machinery, the detached objects no longer exist, and instead the references become orphaned(?). i.e. they point to a dummy object.

**Comment by niklas-ourmachinery** [comment]
@karl-zylinski-ourmachinery I assume there is already a way of dealing with this for textures, etc. Can we copy that workflow for animations?

**Comment by lachsinc** [comment]
IMO copying the 'Extract Assets' ref tracking approach would be nice so that (somewhere down the line) animation extraction settings could be associated w/ the asset e.g. extract to animation subfolder/compress automatically using specified compression settings etc.

Worth noting though users can currently:
- Create new anims/clips and assign the ref manually
- Right click dcc_asset in browser and create clips/animations from the assets with animations

So those actions would need to add themselves to the dcc_asset's list of extracted animations refs, and remove themselves when a user selects a different one.

**Comment by karl-zylinski-ourmachinery** [comment]
I think I could build something that, when the DCC asset reimport happens, just iterates over all objects of type `TM_TT_TYPE__ANIMATION_CLIP` and if the referenced animation belongs to the updated DCC asset then I patch it up by comparing names of animations in the pre and post-update DCC asset.

**Comment by lachsinc** [comment]
Yeah should work. Simple animation player and compressed animation can directly reference dcc animations too so would need to do the same for them.
Arguably simple player could be restricted to only allow selection of .clip's (probably still want to allow playing of dcc anims directly for preview purposes).

**Comment by lachsinc** [comment]
Also with that approach I'm not sure user plugins would have a way of knowing dcc subobjects have changed (aside from polling dcc tt `version()` and checking for id changes) - though that may be me inventing non-existent problems!
